### PR TITLE
update feature top-sell + top-new + count-number

### PR DESCRIPTION
**Pull request by duyminh256**

### CODE REVIEW

File: models/courses.js [hunk 1]
```diff
@@ -19,10 +19,12 @@ const Courses = sequelize.define('courses', {
     type: Sequelize.STRING
```
Comment (hminh21) [inline models/courses.js]: Xoá file này

File: src/course/courseController.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+const NAMESPACE = `courseController-${moment.utc().toISOString()}`
+
+export const getTopNewCourse = async (req, res, next) => {
+  try {
+    const { limit, page } = req.body
+    const topNewCourseList = await getTopNewCourseList(page, limit)
+    if (topNewCourseList) {
+      return res.status(httpStatusCodes.OK).json({
+        message: httpStatusCodes.getStatusText(httpStatusCodes.OK),
```
Comment (hminh21) [inline src/course/courseController.js]: enums.MESSAGE.OK

File: src/course/courseController.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    })
+  }
+}
+export const getTopSellCourse = async (req, res, next) => {
+  try {
+    const { limit, page } = req.body
+    const topSellCourseList = await getTopSellCourseList(page, limit)
+    if (topSellCourseList) {
+      return res.status(httpStatusCodes.OK).json({
+        message: httpStatusCodes.getStatusText(httpStatusCodes.OK),
```
Comment (hminh21) [inline src/course/courseController.js]: enums.MESSAGE.OK

File: src/course/courseController.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      message: 'Hệ thống gặp lỗi khi lấy top khóa học bán được'
+    })
+  }
+}
+export const getCourseNumber = async (req, res, next) => {
+  try {
+    const num = await getNumberOfCourse()
+    if (num) {
+      return res.status(httpStatusCodes.OK).json({
+        message: httpStatusCodes.getStatusText(httpStatusCodes.OK),
```
Comment (hminh21) [inline src/course/courseController.js]: enums.MESSAGE.OK

File: src/course/courseService.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  const instance = await Course.findAll(
+    {
+      where: {
+        isDeleted: false
+      },
+      order: [
+        ['createdAt', 'DESC']
+      ],
+      offset,
+      limit
```
Comment (hminh21) [inline src/course/courseService.js]: raw: true

File: src/course/courseController.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import { debug } from '../utils'
+
+import {
+  getTopNewCourseList,
+  getTopSellCourseList,
+  getNumberOfCourse
+} from './courseService'
+
+const NAMESPACE = `courseController-${moment.utc().toISOString()}`
+
```
Comment (hminh21) [inline src/course/courseController.js]: comment param, body, response bla bla

File: src/course/courseController.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  }
+  catch (err) {
+    debug.error(NAMESPACE, 'Error occured while getting new courses', err)
+    return res.status(httpStatusCodes.INTERNAL_SERVER_ERROR).json({
+      message: 'Hệ thống gặp lỗi khi lấy khóa học mới'
+    })
+  }
+}
+export const getTopSellCourse = async (req, res, next) => {
```
Comment (hminh21) [inline src/course/courseController.js]: comment param, body, response bla bla

File: src/course/courseService.js [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+  const instance = await Course.findAll(
+    {
+      where: {
+        isDeleted: false
+      },
+      order: [
+        ['soldNumber', 'DESC']
+      ],
+      offset,
+      limit
```
Comment (hminh21) [inline src/course/courseService.js]: raw: true

File: models/courses.js [hunk 9]
```diff
@@ -19,10 +19,12 @@ const Courses = sequelize.define('courses', {
...
   },
   requirement: {
-    type: Sequelize.ARRAY(Sequelize.STRING)
+    type: Sequelize.ARRAY(Sequelize.STRING),
+    default: []
   },
   learnWhat: {
-    type: Sequelize.ARRAY(Sequelize.STRING)
+    type: Sequelize.ARRAY(Sequelize.STRING),
+    default: []
```
Comment (nhoknguyen00) [inline models/courses.js]: defaultValue

File: models/courses.js [hunk 10]
```diff
@@ -19,10 +19,12 @@ const Courses = sequelize.define('courses', {
     type: Sequelize.STRING
   },
   requirement: {
-    type: Sequelize.ARRAY(Sequelize.STRING)
+    type: Sequelize.ARRAY(Sequelize.STRING),
+    default: []
```
Comment (nhoknguyen00) [inline models/courses.js]: defaultValue

**Pull request merged**